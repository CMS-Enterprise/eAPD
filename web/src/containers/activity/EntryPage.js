import React from 'react';
import { Switch, Route, useParams, useRouteMatch } from 'react-router-dom';

import ContractorResources from './ContractorResources';
import CostAllocate from './CostAllocate';
import Costs from './Costs';
import Milestones from './Milestones';
import Overview from './Overview';
import Objectives from './Objectives';
import Schedule from './Schedule';
import StandardsAndConditions from './StandardsAndConditions';

const EntryPage = () => {
  const activityIndex = +useParams().activityIndex;

  const { path } = useRouteMatch();

  return (
    <div id="activity-entry-page">
      <Switch>
        <Route path={`${path}/overview`}>
          <Overview activityIndex={activityIndex} />
          <StandardsAndConditions activityIndex={activityIndex} />
        </Route>
        <Route path={`${path}/okrs`}>
          <Objectives activityIndex={activityIndex} />
          <Schedule activityIndex={activityIndex} />
          <Milestones activityIndex={activityIndex} />
        </Route>
        <Route path={`${path}/state-costs`}>
          <Costs activityIndex={activityIndex} />
        </Route>
        <Route path={`${path}/contractor-costs`}>
          <ContractorResources activityIndex={activityIndex} />
        </Route>
        <Route path={`${path}/cost-allocation`}>
          <CostAllocate activityIndex={activityIndex} />
        </Route>
        <Route>
          <Overview activityIndex={activityIndex} />
          <StandardsAndConditions activityIndex={activityIndex} />
        </Route>
      </Switch>{' '}
    </div>
  );
};

export default EntryPage;
