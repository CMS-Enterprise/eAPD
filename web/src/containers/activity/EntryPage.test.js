import { shallow } from 'enzyme';
import React from 'react';

import { EntryPage, mapStateToProps } from './EntryPage';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ activityIndex: 2 })
}));

describe('single activity entry page', () => {
  it('renders correctly', () => {
    expect(
      shallow(<EntryPage activityNames={['A1', 'A2', 'A3', 'A4']} />)
    ).toMatchSnapshot();
  });

  it('maps redux state to properties', () => {
    expect(
      mapStateToProps({
        apd: {
          data: {
            activities: [{ name: 'one' }, { name: 'two' }, { name: 'three' }]
          }
        }
      })
    ).toEqual({ activityNames: ['one', 'two', 'three'] });
  });
});
