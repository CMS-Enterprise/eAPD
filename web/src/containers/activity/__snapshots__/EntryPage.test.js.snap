// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single activity entry page renders correctly 1`] = `
<div
  id="activity-entry-page"
>
  <h2>
    Activity 
    3
    : 
    A3
  </h2>
  <ActivityRoutes
    activityIndex={2}
  />
</div>
`;
